File: problemsets/Codeforces/Python/B785.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
n=int(input())
a=sorted([[*map(int,input().split())]for _ in[0]*n])
m=int(input())
b=sorted([[*map(int,input().split())]for _ in[0]*m])
c=sorted(a,key=lambda x:x[1])
d=sorted(b,key=lambda x:x[1])
print(max(max(0,a[n-1][0]-d[0][1]),max(0,b[m-1][0]-c[0][1])))

File: problemsets/Codeforces/C++/B785.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include <bits/stdc++.h>

using namespace std;

#define eb emplace_back
#define fi first
#define se second
typedef pair<int,int> pii;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n, m, minA, minB, maxA, maxB;
	minA = minB = INT_MAX;
	maxA = maxB = INT_MIN;
	cin >> n;
	for (int i = 0; i < n; i++) {
	  int x, y; cin >> x >> y;
	  minA = min(y, minA);
	  maxA = max(x, maxA);
	}
	cin >> m;
	for (int i = 0; i < m; i++) {
	  int x, y; cin >> x >> y;
	  minB = min(y, minB);
	  maxB = max(x, maxB);
	}
  cout << max({0,maxA-minB,maxB-minA});
}